adapters/src/main/java/io/realm/RealmCursor.java

```diff
@@ -377,11 +375,8 @@ public int getInt(int columnIndex) {
         switch (columnType) {
             case INTEGER:
                 long aLong = currentRow.getLong(columnIndex);
-                if (aLong <= Integer.MIN_VALUE)
-                    return Integer.MIN_VALUE;
-                if (aLong >= Integer.MAX_VALUE)
-                    return Integer.MAX_VALUE;
-                return (int) aLong;
+                if (aLong > Integer.MIN_VALUE && aLong < Integer.MAX_VALUE)
```

Comment:
I think this code path will throw with `canNotConvert()` since there is no `break`.